pkgs/development/tools/build-managers/gradle/default.nix

```diff
@@ -15,7 +15,29 @@ rec {
         --add-flags "-classpath $gradle_launcher_jar org.gradle.launcher.GradleMain"
     '';
 
-    phases = "unpackPhase installPhase";
+    libnativeFixup =
+      let
+        arch =
+          { "x86_64-linux" = "linux-amd64";
+            "i686-linux" = "linux-i386";
+          }."${stdenv.system}" or "";
+       in ''
+        mkdir patching
+        pushd patching
+        jar xf $out/lib/gradle/lib/native-platform-${arch}-0.10.jar
+        ${patchelf}/bin/patchelf --set-rpath "${stdenv.cc.cc}/lib:${stdenv.cc.cc}/lib64" net/rubygrapefruit/platform/${arch}/libnative-platform.so
+        jar cf native-platform-${arch}-0.10.jar .
+        mv native-platform-${arch}-0.10.jar $out/lib/gradle/lib/
+        popd
+
+        # The scanner doesn't pick up the runtime dependency in the jar.
+        # Manually add a reference where it will be found.
+        mkdir $out/nix-support
+        echo ${stdenv.cc.cc} > $out/nix-support/manual-runtime-dependencies
+      '';
+
+    fixupPhase = if stdenv.isLinux then libnativeFixup else "true";
```

Comment:
Oh, I see --- libnativeFixup still gets interpolated and exported to the builder.